src/warp.jl

```diff
@@ -0,0 +1,38 @@
+# None of these types are provided by this package, so the following line seems unclean
+#     @inline Base.getindex(A::AbstractExtrapolation, v::StaticVector) = A[convert(Tuple, v)...]
+# furthermore it would be ambiguous with getindex(::Extrapolation, xs...) after https://github.com/tlycken/Interpolations.jl/pull/141
+@inline _getindex(A, v::StaticVector) = A[convert(Tuple, v)...]
+
+warp(img::AbstractArray, args...) = warp(interpolate(img, BSpline(Linear()), OnGrid()), args...)
+
+function warp{T<:Colorant,S}(::Type{T}, img::AbstractArray{S}, args...)
+    TCol = typeof(convert(T,one(S)))
+    TNorm = eltype(TCol)
+    apad, pad = Interpolations.prefilter(TNorm, TCol, img, typeof(BSpline(Linear())), typeof(OnGrid()))
+    itp = Interpolations.BSplineInterpolation(TNorm, apad, BSpline(Linear()), OnGrid(), pad)
+    warp(itp, args...)
+end
```

Comment:
The way this function works may or may not be what we want. I am unsure. It is quite flexible though.

Some examples:

```julia
warp(RGB, ::Array{<:Colorant{N0f8}}) --> Array{RGB{N0f8}} 
warp(RGB{Float32}, ::Array{<:Colorant{N0f8}}) --> Array{RGB{Float32}} 
warp(Gray, ::Array{<:Colorant{N0f8}}) --> Array{Gray{N0f8}} 
warp(Gray, ::Array{<:Colorant{Float64}}) --> Array{Gray{Float64}} 
warp(Gray{N0f8}, ::Array{<:Colorant{Float64}}) --> Array{Gray{N0f8}} 
```